Release notes 2.15.0 (#3186)
timescale/docs: about/release-notes/index.md

click `Watch`, select `Custom` and then check `Releases`.
</Highlight>

## TimescaleDB&nbsp;2.15.0 on 2024-05-08

<Highlight type="note">

> This release contains performance improvements and bug fixes since

Reviewer: ```suggestion
This release contains performance improvements and bug fixes introduced since
TimescaleDB v2.14.2. Best practice is to upgrade at the next available opportunity.
```

the 2.14.2 release. We recommend that you upgrade at the next
available opportunity.
</Highlight>

#### Highlighted features in this release

> * Support `time_bucket` with `origin` and/or `offset` on Continuous Aggregate

Reviewer: ```suggestion
* [Continuous Aggregate](/api/:currentVersion:/continuous-aggregates/create_materialized_view/#sample-usage) now support `time_bucket` with `origin` and/or `offset`.
```

> * Compression improvements:

Reviewer: ```suggestion
* The following improvements have been introduced for [hypertable compression](/api/:currentVersion:/compression/):
```

>   - Improve expression pushdown

Reviewer: Expression pushdown is improved. 

I can't find the term "expression pushdown" in the docs or in google. Is there another way to say this?

Author: Rephrased it a bit!

>   - Add minmax sparse indexes when compressing columns with btree indexes

Reviewer: ```suggestion
  - You can now use `minmax` sparse indexes when you compress columns with `btree` indexes
  - Compression now uses the default functions
  - Filters in WHERE clauses that contain text equality operators and LIKE expressions are now vectorized.
```

  - Make compression use the defaults functions
  - Vectorize filters in WHERE clause that contain text equality operators and LIKE expressions

#### Deprecation warning

> * Starting on this release will not be possible to create Continuous Aggregate using `time_bucket_ng` anymore and it will be completely removed on the upcoming releases.

Reviewer: ```suggestion
*  You can no longer create continuous aggregates using `time_bucket_ng`. Best practice is to [migrate your current continuous aggregates to the new format](/use-timescale/:currentVersion:/continuous-aggregates/migrate/), this feature will be completely removed the next release. 
```

> * Recommend users to [migrate their old Continuous Aggregate format to the new one](https://docs.timescale.com/use-timescale/latest/continuous-aggregates/migrate/) because it support will be completely removed in next releases prevent them to migrate.

Reviewer: Remove this line, see comment above.

* This is the last release supporting PostgreSQL 13.

> #### For on-premise users and this release only

Reviewer: ```suggestion
#### For self-hosted deployments of TimescaleDB v2.15.0 only
```

> You will need to run [this SQL script](https://github.com/timescale/timescaledb-extras/blob/master/utils/2.15.X-fix_hypertable_foreign_keys.sql) after running `ALTER EXTENSION`. More details can be found in the pull request [#6786](#6797).

Reviewer: ```suggestion
After you run `ALTER EXTENSION`, you must run [this SQL script](https://github.com/timescale/timescaledb-extras/blob/master/utils/2.15.X-fix_hypertable_foreign_keys.sql). For more details, see the following pull requests [#6786](#6797).
```

* #6617 Fix error in show_chunks
* #6621 Remove metadata when dropping chunks
* #6677 Fix snapshot usage in CAgg invalidation scanner
* #6698 Define meaning of 0 retries for jobs as no retries
* #6717 Fix handling of compressed tables with primary or unique index in COPY path
* #6726 Fix constify cagg_watermark using window function when querying a CAgg
* #6729 Fix NULL start value handling in CAgg refresh
* #6732 Fix CAgg migration with custom timezone / date format settings
* #6752 Remove custom autovacuum setting from compressed chunks
* #6770 Fix plantime chunk exclusion for OSM chunk
* #6789 Fix deletes with subqueries and compression
* #6796 Fix a crash involving a view on a hypertable
* #6797 Fix foreign key constraint handling on compressed hypertables
* #6816 Fix handling of chunks with no contraints
* #6820 Fix a crash when the ts_hypertable_insert_blocker was called directly
* #6849 Use non-orderby compressed metadata in compressed DML
* #6867 Clean up compression settings when deleting compressed cagg
* #6869 Fix compressed DML with constraints of form value OP column
* #6870 Fix bool expression pushdown for queries on compressed chunks

### Acknowledgments

> * @brasic for reporting a crash when the ts_hypertable_insert_blocker was called directly

Reviewer: ```suggestion
* @brasic for reporting a crash when `ts_hypertable_insert_blocker` was called directly
```

* @bvanelli for reporting an issue with the jobs retry count

> * @djzurawsk For reporting error when dropping chunks

Reviewer: ```suggestion
* @djzurawsk for reporting an error about dropping chunks
```

> * @Dzuzepppe for reporting an issue with DELETEs using subquery on compressed chunk working incorrectly.

Reviewer: ```suggestion
* @Dzuzepppe for reporting the issue that DELETE using a subquery on compressed chunks was working incorrectly.
```

* @hongquan For reporting a 'timestamp out of range' error during CAgg migrations

> * @kevcenteno for reporting an issue with the show_chunks API showing incorrect output when 'created_before/created_after' was used with time-partitioned columns.

Reviewer: ```suggestion
* @kevcenteno for reporting that the `show_chunks` API returned an incorrect output when 'created_before/created_after' was used with time-partitioned columns.
```

* @mahipv For starting working on the job history PR

> * @rovo89 For reporting constify cagg_watermark not working using window function when querying a CAgg

Reviewer: ```suggestion
* @rovo89 For reporting that constify `cagg_watermark` was not working using the window function when querying a CAgg.
```
